src/MongoDB/ServerDescription.c

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright 2021-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <php.h>
+#include <Zend/zend_interfaces.h>
+#include <ext/standard/php_var.h>
+#include <zend_smart_str.h>
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "phongo_compat.h"
+#include "php_phongo.h"
+
+zend_class_entry* php_phongo_serverdescription_ce;
+
+/* {{{ proto array MongoDB\Driver\ServerDescription::getHelloResponse()
+   Returns the most recent “hello” response */
+static PHP_METHOD(ServerDescription, getHelloResponse)
+{
+	php_phongo_serverdescription_t* intern;
+	const bson_t*                   helloResponse;
+
+	intern = Z_SERVERDESCRIPTION_OBJ_P(getThis());
+
+	PHONGO_PARSE_PARAMETERS_NONE();
+
+	helloResponse = mongoc_server_description_hello_response(intern->server_description);
+
+	if (helloResponse->len) {
```

Comment:
Ah ok, I got it from the `getTagSets()` function in `ReadPreference`. Why is it used there?
https://github.com/mongodb/mongo-php-driver/blob/80f88146f82d5d3f52be16b60742820b2b55abdb/src/MongoDB/ReadPreference.c#L456

Reply:
Looks like a mistake dating back to the first version of the driver (https://github.com/mongodb/mongo-php-driver/commit/27ee04ff888ed5e86680fef5fa3709f4c6a3a587). Would you mind opening a separate ticket as a reminder to clean that up?

Doing so wouldn't require any API changes as the method is [documented](https://www.php.net/manual/en/mongodb-driver-readpreference.gettagsets.php) as only returning an array. So that's just dead code to be removed.

Reply:
https://jira.mongodb.org/browse/PHPC-1927